dev/com.ibm.ws.kernel.boot.common/src/io/openliberty/checkpoint/spi/CheckpointPhase.java

```diff
@@ -125,6 +126,35 @@ static synchronized void setPhase(String p) {
             }
         }
         THE_PHASE = phase;
+        if (phase != INACTIVE) {
+            phase.addMultiThreadedHook(new CheckpointHook() {
```

Comment:
I should add a rank here so we obtain the write lock as late as possible before going into single-thread mode.

Reply:
fixed